.github/workflows/main.yml

```diff
@@ -0,0 +1,48 @@
+name: Main pipeline for linting and testing
+
+on: [push]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Check out
+        uses: actions/checkout@v4
+
+      - name: Set up Python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10.*"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .[dev]
+
+      - name: Run pre-commit
+        # FIXME: Does only run on subset of files since not all files are linted yet
+        run: pre-commit run --files pyproject.toml .pre-commit-config.yaml requirements/* --show-diff-on-failure
+
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        python-version: ["3.10.*"]
```

Comment:
minor suggestion: you should add more python versions given that the readme states that Qrisp is compatible with 3.10, 3.11 & 3.12.